### PR TITLE
[Core/Misc]: Restore playercreateinfo_item item deletion functionality.

### PR DESCRIPTION
**Pull request by lineagedr**
## Changes Proposed:
-  Partially revert https://github.com/azerothcore/azerothcore-wotlk/pull/3122

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/5919

## Tests Performed:

- Tested using the method below

## How to Test the Changes:

1. Create a fresh DK character.
2. Open your bags
3. Look if you have the item ```[TDB PH] - unsused Scourgestone```
4. If you don't see the item in your inventory then the PR did it's job

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by benjymansy123** [comment]
Looks like "Scourge" is misspelled in your SQL update.

**Comment by lineagedr** [comment]
> Looks like "Scourge" is misspelled in your SQL update.

thanks

File: data/sql/updates/pending_db_world/rev_1629485842027723200.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1629485842027723200');
+
+ALTER TABLE `playercreateinfo_item`
+    CHANGE `amount` `amount` TINYINT(4) SIGNED NOT NULL DEFAULT 1;
```
[the remark this answers is not in the document]
Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1629485842027723200.sql:4]: Some stuff can stack to 1000

File: data/sql/updates/pending_db_world/rev_1629485842027723200.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+
+INSERT INTO `playercreateinfo_item` (`race`, `class`, `itemid`, `amount`, `Note`) VALUES (0, 6, 40582, -1, "[TDB PH] - unsused Scourgestone");
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1629485842027723200.sql:6]: Always delete before insert

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1629485842027723200.sql [hunk 3]
```diff
@@ -1,6 +1,7 @@
 INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1629485842027723200');
 
 ALTER TABLE `playercreateinfo_item`
-    CHANGE `amount` `amount` TINYINT(4) SIGNED NOT NULL DEFAULT 1;
+    CHANGE `amount` `amount` INT(11) SIGNED NOT NULL DEFAULT 1;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1629485842027723200.sql:4]:
```suggestion
    CHANGE `amount` `amount` INT SIGNED NOT NULL DEFAULT 1;
```

**Review by Kitzunu — CHANGES_REQUESTED** [review]
Int width is deprecated or will be deprecated in later versions of MySQL 8.0 don't use it

File: data/sql/updates/pending_db_world/rev_1629485842027723200.sql (hunk 3 above)
Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1629485842027723200.sql:4]: Also you dont really need anything bigger than SMALLINT

Comment (lineagedr) [reply data/sql/updates/pending_db_world/rev_1629485842027723200.sql:4]: @Kitzunu We do, look at the C++ code.

**Review by Kitzunu — APPROVED** [review]

**Comment by pklloveyou** [comment]
Tested

**Pull request merged**